src/resmom/mom_main.c

```diff
@@ -8742,15 +8742,14 @@ main(int argc, char *argv[])
 			LOG_WARNING, msg_daemonname, winlog_buffer);
 	}
 #endif
+
 	/*
-	 * Set mom_host to PBS_MOM_NODE_NAME if it is defined.
-	 * Otherwise, set mom_host with a call to gethostname().
+	 * Query the hostname.
+	 * Set mom_host temporarily with a call to gethostname().
 	 */
-	c = 0;
-	if (pbs_conf.pbs_mom_node_name) {
-		/* Hostname was defined by PBS_MOM_NODE_NAME */
-		(void)strncpy(mom_host, pbs_conf.pbs_mom_node_name, (sizeof(mom_host) - 1));
-		mom_host[(sizeof(mom_host) - 1)] = '\0';
+	c = gethostname(mom_host, (sizeof(mom_host) - 1));
+	if (c != 0) {
```

Comment:
Did you mean if (c == 0) which is the sucess return of gethostname()?

Reply:
Yes, good catch. Fixed. I'll redo the tests too.